update security considerations (#38)
VMatrix1900/draft-cats-metric-definition: draft-ietf-cats-metric-definition.md

SEC-1: All metric messages MUST be protected with cryptographic integrity mechanisms. Receivers MUST verify the integrity of every metric message before using enclosed metrics for traffic steering decisions. Any detected integrity violation MUST cause the relevant metric message to be discarded and SHOULD trigger an alert absent local policy.

*Authenticity*: ensures that a metric message genuinely originates from the claimed service contact instance or a trusted publisher.

SEC-2: Each metric publisher MUST be authenticated using strong cryptographic credentials. Metric messages MUST carry a verifiable proof of origin that binds a metric to the publisher's identity. Receivers MUST reject metric messages that cannot be authenticated, regardless of their content.

*Controllability*: ensures that only explicitly authorized entities are allowed to publish metrics.

SEC-3: A CATS system MUST enforce fine-grained control policies that authorize which publishers can report metrics for which services. Unauthorized publication attempts MUST be rejected and logged.

> *Freshness*: ensures that metric values are not stale or replayed. Only sufficiently recent metric values are used for decision-making.

Reviewer: Did we decide to add a timestamps parameter to the metrics so the receiver can decide whether the metric is stale? If so, suggest mentioning it here too.

Author: Med commented that mentioning timestamp explicitly here has implication on time sycnhronization that need to be called out. It's better left for solutions and we can keep the discussion to be general here.

# Security Considerations

> CATS metrics are not merely telemetry data, but are important inputs to the traffic steering selection logic. Incorrect or manipulated metrics can directly affect forwarding behaviour, potentially leading to service disruption or policy violation. CATS metrics are dynamic and sensitive. They may indirectly expose physical locations, health status, or service topology information of CATS service instances. Attackers could sample or correlate metrics to infer internal service instance states and mount targeted attacks. Therefore, the security properties of CATS metric - integrity, authenticity, controllability, freshness, and confidentiality, are of critical importance.

Reviewer 2: ```suggestion
CATS metrics are not merely telemetry data, but are important inputs to the traffic steering selection logic. Similar to routing metrics, incorrect or manipulated metrics can directly affect the underlying forwarding behaviour, potentially leading to service disruption, denial-of-service, or policy violation. CATS metrics are dynamic and sensitive. They may indirectly expose physical locations, health status, or service topology information of CATS service contact instances. Access to such metric would allow attackers to sample or correlate metrics to infer internal service instance states and mount targeted attacks. Therefore, the security properties of CATS metrics - integrity, authenticity, controllability, freshness, and confidentiality, are of critical importance. Adequate measures are required to prevent these metrics are exposed to non authorized entites.
```

*Integrity*: ensures that metric values have not been altered by unauthorized entities during collection, distribution, or storage.

> SEC-1: All metric messages MUST be protected with cryptographic integrity mechanisms. Receivers MUST verify the integrity of every metric before using it for steering decisions. Any detected integrity violation MUST cause the metric to be discarded and SHOULD trigger an alert.

Reviewer 2: ```suggestion
SEC-1: All metric messages MUST be protected with cryptographic integrity mechanisms. Receivers MUST verify the integrity of every metric message before using enclosed metrics for traffic steering decisions. Any detected integrity violation MUST cause the relevant metric message to be discarded and SHOULD trigger an alert absent local policy.
```

> *Authenticity*: ensures that a metric genuinely originates from the claimed service instance or trusted publisher.

Reviewer 2: ```suggestion
*Authenticity*: ensures that a metric message genuinely originates from the claimed service contact instance or a trusted publisher.
```

Metrics are not issued by the service instance themselves (as that is hiddern to CATS). I would reword.

> SEC-2: Each metric publisher MUST be authenticated using strong cryptographic credentials. Metrics MUST carry a verifiable proof of origin that binds the metric to the publisher's identity. Receivers MUST reject metrics that cannot be authenticated, regardless of their content.

Reviewer 2: ```suggestion
SEC-2: Each metric publisher MUST be authenticated using strong cryptographic credentials. Metric messages MUST carry a verifiable proof of origin that binds a metric to the publisher's identity. Receivers MUST reject metric messages that cannot be authenticated, regardless of their content.
```

several metric values may be carried in a single message

> *Controllability*: ensures that only explicitly authorized entities are allowed to publish metrics.

Reviewer 2: The doc should make a distinction beween metric and metric value

Author: Thanks. I will go through the entire draft to distinguish between metric and metric value, after updating the sec considerations.

> SEC-3: The CATS system MUST enforce fine-grained control policies that authorize which publishers can report metrics for which service instances. Unauthorized publication attempts MUST be rejected and logged.

Reviewer 2: ```suggestion
SEC-3: A CATS system MUST enforce fine-grained control policies that authorize which publishers can report metrics for which services. Unauthorized publication attempts MUST be rejected and logged.
```

> *Freshness*: ensures that metrics are not stale or replayed; only sufficiently recent metrics are used for decision-making.

Reviewer 2: ```suggestion
*Freshness*: ensures that metric values are not stale or replayed; only sufficiently recent metric values are used for decision-making.
```

> SEC-4: Each metric MUST include a precise timestamp and SHOULD include an explicit validity interval or lifetime. Receivers and C-PS MUST enforce freshness checks.

Reviewer 2: ```suggestion
SEC-4: Each metric message MUST include a precise timestamp and SHOULD include an explicit validity interval or lifetime. Receivers and C-PS MUST enforce freshness checks.
```

This has implication on time sycnhronization that need to be called out.

Do we reallly need to have one approach called it for replay prevention? Why not leave that to solutions and focus on the main requirement?

Author: How about a more general statement:
"SEC-4: Each metric message MUST be protected against replay and stale value. Receivers and C-PS MUST enforce freshness checks.
"
@boucadair Hi Med, how do you think of the above changes?

Reviewer 2: This is better. Thanks.

*Confidentiality*: ensures that metric data is not disclosed to unauthorized entities during transmission or storage.

> SEC-5: Sensitive metric fields SHOULD be encrypted. Encryption keys MUST be managed securely and rotated periodically. Even when metrics are aggregated, techniques such as differential privacy SHOULD be used to prevent inference of sensitive information.

Reviewer 2: what is a "sensitive metrcic field"? When it is not required to encrypt it?

Author: Thanks, Med. I agree with you that the above statement is ambiguous. How about the following changes:
"
*Confidentiality*: ensures that metric values are not disclosed to unauthorized entities during transmission or storage.

SEC-5: Metric messages SHOULD be encrypted in transit, as they may contain sensitive information. Encryption keys MUST be managed securely and rotated periodically. Even when metric values are aggregated, techniques such as differential privacy SHOULD be used to prevent inference of sensitive information.
"

Reviewer 2: > Metric messages SHOULD be encrypted in transit, as they may contain sensitive information

It is still not clear when it is OK to not encrypt in transit. BTW, transit may be misinterpretd as this is exiting an administrative domain, while this is not. 

> such as differential privacy SHOULD be used to prevent inference of sensitive information.

* Add a reference to "differential privacy"
* I think that you can get rid of "SHOULD" here and use "may" or "should" as this is about an example

Author: Thanks! Yours is more rigorous.
